### ISSUE TITLE
`author` metadata set in `_metadata.yml` and in YAML header inside document are not correctly merged. 

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#9864]
### Discussed in https://github.com/quarto-dev/quarto-cli/discussions/9855

<div type='discussions-op-text'>

<sup>Originally posted by **McKabue** June  3, 2024</sup>

**/posts/welcome/index.qmd**

```
---
title: "Welcome To My Blog"
date: "2024-05-01"
categories: [news]
author: "martin from welcome"
---

This is the first post in a Quarto blog. Welcome!

![](thumbnail.jpg)

Since this post doesn't specify an explicit `image`, the first image in the post will be used in the listing page of posts.
```

**/posts/_metadata.yml**
```
categories: [tech]
author:
  - name: James from posts
```

With the above setup, I get the following post and listing:

<img width="674" alt="Screenshot 2024-06-03 at 22 21 52" src="https://github.com/quarto-dev/quarto-cli/assets/6072072/aadea43a-4875-40a5-bba2-a0667f2c2bfa">
<img width="939" alt="Screenshot 2024-06-03 at 22 22 14" src="https://github.com/quarto-dev/quarto-cli/assets/6072072/55e9e0c6-ad6c-4995-a3c2-1f11ab6fe0c6">

*****

However, if:

**/posts/welcome/index.qmd**

```
---
title: "Welcome To My Blog"
date: "2024-05-01"
---

This is the first post in a Quarto blog. Welcome!

![](thumbnail.jpg)

Since this post doesn't specify an explicit `image`, the first image in the post will be used in the listing page of posts.
```

**/posts/welcome/_metadata.yml**
```
categories: [news]
author: "martin from welcome"
```

**/posts/_metadata.yml**
```
categories: [tech]
author:
  - name: James from posts
```

I get the following post and listing:

<img width="939" alt="Screenshot 2024-06-03 at 22 25 43" src="https://github.com/quarto-dev/quarto-cli/assets/6072072/cef4bb57-738b-4f39-ac0a-30097b08566c">
<img width="939" alt="Screenshot 2024-06-03 at 22 25 54" src="https://github.com/quarto-dev/quarto-cli/assets/6072072/5dcdb8b7-08aa-40ac-a904-068a3dd487d3">

*****

When the metadata is in the document file, it overwrites all parent metadata such as authors and categories instead of extending it in the post page.

</div>

**Comment by cderv** [comment]
I looked quickly into this to diagnose and I think this happens also in regular rendering not only for blog or listing. 

Basically, we have a part in our code where we "override" any processing that could have been done, by forcing the use of the metadata inside the document. 
https://github.com/quarto-dev/quarto-cli/blob/b34b24d924f1d7b708611592c0d96fd8b3adc3a3/src/command/render/pandoc.ts#L978-L1002

So basically in here: 
- `pandocMetadata['author']` has correctly the merged authors so the two value
- But `author` key is not identified as a quarto metadata and so it gets overriden by `engineMetadata['author'];` which is the document value set. 

We got already bitten by this in revealjs for themes, where I added an escape
https://github.com/quarto-dev/quarto-cli/blob/b34b24d924f1d7b708611592c0d96fd8b3adc3a3/src/command/render/pandoc.ts#L994-L998

This is very old codebase (341cc3b8eb79c6cc53e172dec3b532148db107c9 from 3 years) for backward compatibility for inline yaml rendering. 

So I think either we keep it but we probably need to consider `author` as a `isQuartoMetadata` key because we process it. Or we should remove this overwrite which does not make sense anymore probably.

**Comment by mcanouil** [comment]
FYI:
- https://github.com/quarto-dev/quarto-cli/discussions/8860

it was another issue were all were merged resulting in duplicates, the fix was: https://github.com/quarto-dev/quarto-cli/commit/478b3df44b6db1e97d9bb25d5ec4dc6b0f33cdb8

**Comment by cderv** [comment]
Yes this is different. #8860 is really about listing. As I said here, the problem happening also with default project, using shared metadata.

**Comment by cderv** [comment]
A note that this is not only for `author`, but any `key` that won't be identified in `!isQuartoMetadata(key) && !isChapterTitle && !isIncludeMetadata(key)`

**Comment by ute** [comment]
I had the same problem when I tried to update keys for extensions, that is, I have some definitions in `_metadata.yml` and others in the document yaml. Here is a workaround for the time being (just if someone else needs it in the meanwhile): [mergmeta extension](https://github.com/ute/mergemeta). It actually allows some kind of manually controlled cascading, because the top level keys that should be merged need to have distinct names.
